### PR TITLE
improve structure of config and manifest reference

### PR DESCRIPTION
**Pull request by jluebbe**
Using titles makes the index more useful and allows us to deep-link to a specific section. 

Sometimes, we noted required/optional keys in the description, but it's better make it easy to see.

### CODE REVIEW

File: docs/reference.rst [hunk 1]
```diff
@@ -152,7 +152,7 @@ Example configuration:
 
 .. _data-directory:
 
-``data-directory``
+``data-directory`` (optional, recommended)
```
Comment (ejoerns) [inline docs/reference.rst:155]: For all others, we have either the one or the other.

File: docs/reference.rst [hunk 2]
```diff
@@ -424,17 +423,17 @@ See details about using handlers in `Custom Handlers (Interface)`_.
...
   bundle content.
 
-``post-install``
+``post-install`` (optional)
   This handler will be called after a successful installation.
   The bundle is still mounted at this moment, thus you could access data in it
   if required.
 
-``bootloader-custom-backend``
+``bootloader-custom-backend`` (optional)
```
Comment (ejoerns) [inline docs/reference.rst:436]: It depends. It's mandatory for `bootloader = custom`.

File: docs/reference.rst [hunk 3]
```diff
@@ -739,30 +738,30 @@ unique per ``system.conf``.
...
-``filename``
+``filename`` (required)
   The log file name used for logging.
   If no absolute path is given, the location is assumed to be relative to the
   ``data-directory``.
   Using a relative file name without ``data-directory`` set will cause a
   configuration error.
 
-``events``
+``events`` (required)
```
Comment (ejoerns) [inline docs/reference.rst:748]: Optional since it defaults to `"all"`.

File: docs/reference.rst [hunk 4]
```diff
@@ -739,30 +738,30 @@ unique per ``system.conf``.
...
   Semicolon-separated list of events to log. Currently supported event types are:
 
   * ``install`` - Logs start and end of installation
   * ``boot`` - Logs boot information
   * ``mark`` - Logs slot marking information
   * ``all`` - Log all events (cannot be combined with other events)
 
-``format``
+``format`` (required)
```
Comment (ejoerns) [inline docs/reference.rst:756]: Optional, since it defaults to `"readable"`.

**Review by ejoerns — CHANGES_REQUESTED** [review]

File: docs/reference.rst (hunk 1 above)
Comment (jluebbe) [reply docs/reference.rst:155]: For the others we have either required or optional. "recommended" is independent of that.

Comment (ejoerns) [reply docs/reference.rst:155]: You mean like

    ``type=<type>`` (recommended)

and others?

Comment (jluebbe) [reply docs/reference.rst:155]: `type=` should be (optional, recommended) as well.

Comment (ejoerns) [reply docs/reference.rst:155]: Fine for me. Same then for `bundle-formats` and `format`.

**Review by ejoerns — APPROVED** [review]

**Pull request merged**